src/groovy/asset/pipeline/grails/AssetPipelineFilter.groovy

```diff
@@ -1,75 +1,83 @@
 package asset.pipeline.grails
 
+
 import asset.pipeline.AssetPipeline
-import asset.pipeline.AssetPipelineConfigHolder
 import asset.pipeline.AssetPipelineResponseBuilder
 import grails.util.Environment
 import groovy.util.logging.Slf4j
+import org.springframework.core.io.Resource
+import org.springframework.web.context.WebApplicationContext
 import org.springframework.web.context.support.WebApplicationContextUtils
 
 import javax.servlet.*
-import java.util.TimeZone
 import java.text.SimpleDateFormat
 
+
 @Slf4j
 class AssetPipelineFilter implements Filter {
-	public static final String HTTP_DATE_FORMAT = "EEE, dd MMM yyyy HH:mm:ss zzz"
-	
-	public static final ProductionAssetCache fileCache = new ProductionAssetCache();
 
-	def applicationContext
-	def servletContext
-	def warDeployed
-	void init(FilterConfig config) throws ServletException {
+	static final String HTTP_DATE_FORMAT = "EEE, dd MMM yyyy HH:mm:ss zzz"
+
+	static final ProductionAssetCache fileCache = new ProductionAssetCache()
+
 
+	WebApplicationContext applicationContext
+	ServletContext        servletContext
+	boolean               warDeployed
+
+
+	@Override
+	void init(final FilterConfig config) throws ServletException {
 		applicationContext = WebApplicationContextUtils.getWebApplicationContext(config.servletContext)
 		servletContext = config.servletContext
 		warDeployed = Environment.isWarDeployed()
 	}
 
+	@Override
 	void destroy() {
 	}
 
-	void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
-		def mapping = applicationContext.assetProcessorService.assetMapping
+	@Override
+	void doFilter(final ServletRequest request, final ServletResponse response, final FilterChain chain) throws IOException, ServletException {
+		final String mapping = applicationContext.assetProcessorService.assetMapping
 
-		def fileUri = new URI(request.requestURI).path
-		def baseAssetUrl = request.contextPath == "/" ? "/$mapping" : "${request.contextPath}/${mapping}"
-		def format = servletContext.getMimeType(fileUri)
-		def encoding = request.getParameter('encoding') ?: request.getCharacterEncoding()
+		String fileUri = new URI(request.requestURI).path
+
+		final String baseAssetUrl = request.contextPath == "/" ? "/$mapping" : "${request.contextPath}/${mapping}"
+		final String format       = servletContext.getMimeType(fileUri)
+		final String encoding     = request.getParameter('encoding') ?: request.getCharacterEncoding()
 		if(fileUri.startsWith(baseAssetUrl)) {
 			fileUri = fileUri.substring(baseAssetUrl.length())
 		}
 		if(warDeployed) {
-			def manifest = AssetPipelineConfigHolder.manifest
-			def manifestPath = fileUri
-			if(fileUri.startsWith('/')) {
-			  manifestPath = fileUri.substring(1) //Omit forward slash
-			}
-			fileUri = manifest?.getProperty(manifestPath, manifestPath)
+			fileUri = applicationContext.assetProcessorService.getAssetPath(fileUri)
```

Comment:
It was intentional that the servlet filter be entirely self sufficient and not utilize any grails specific beans. This is the one area i would not touch with regards to accessing the assetProcessorService